### PR TITLE
chore(divider): added tokens, fixed breaking change comments

### LINKED ISSUE #6136: Divider tokens

### PR DESCRIPTION
**Pull request by mattnolting**
closes #6136 

Couple of things to note
• Added `// * {Component}` to `:root` variable comments to make them searchable by `{Component}` or `* {Component}`
• Added `// - {Component}` to component/elements comments to make them searchable by `{Component}` or `- {Component}`
• Removed unnecessary `::after` styling and breaking change comments
• Simplified implementation

### CODE REVIEW

File: src/patternfly/components/Divider/divider.scss [hunk 1]
```diff
@@ -1,98 +1,83 @@
...
-      calc(
-        var(--#{$divider}--m-vertical--after--FlexBasis) - var(--#{$divider}--after--Inset)
-      ); // update at breaking change
-  }
+  flex-direction: column;
+  width: var(--#{$divider}--Size); // apply size to width in vertical orientation
+  height: 100%;
+}
+
+:root {
```
[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Divider/divider.scss:19]: We did, for specific components that need that support due to mixin inclusions. I don't see a need for that here, as it increases specificity, which could result in unwanted styling collisions

[the remark this answers is not in the document]
Comment (mattnolting) [reply src/patternfly/components/Divider/divider.scss:19]: | increases specificity
@srambach limits the ability to update variables from a parent selector, rather

[the remark this answers is not in the document]
Comment (mcoker) [reply src/patternfly/components/Divider/divider.scss:19]: @mattnolting the plan for now is to update all the components we're working on to use both selectors, just so we don't run into issues from updating the scope, then once we're in a good spot with the updates, update them just to use `:root` and run the screenshot tool before/after and see what all breaks, then go from there. WDYT of that? There may be other scoping issues (other than the mixin bit we saw) we may not catch, but the screenshot tool should catch 'em all.

File: src/patternfly/components/Divider/divider.scss [hunk 2]
```diff
@@ -1,98 +1,83 @@
...
-
-  &::after {
-    flex-basis:
-      calc(
-        var(--#{$divider}--after--FlexBasis) - calc(var(--#{$divider}--after--Inset) * 2)
-      ); // update at breaking change
-  }
+  flex-direction: row;
+  width: 100%;
+  height: var(--#{$divider}--Size);  // apply size to width in horizontal orientation
```
Comment (mcoker) [inline src/patternfly/components/Divider/divider.scss:9]: nit, imo the comments probably aren't necessary

```suggestion
  height: var(--#{$divider}--Size);  // apply size to height in horizontal orientation
```

File: src/patternfly/components/Divider/divider.scss [hunk 3]
```diff
@@ -1,98 +1,83 @@
...
-  border: 0; // removes the default border styling on an hr
 
-  &::after {
-    align-self: stretch;
-    justify-self: center;
-    width: var(--#{$divider}--after--Width);
-    height: var(--#{$divider}--after--Height);
+  &:is(hr) {
+    border: 0; // removes the default border styling on an hr
+  }
```
Comment (mcoker) [inline src/patternfly/components/Divider/divider.scss:40]: just curious why we need to target `hr` and not just use this as a base style? Kinds of like how we apply `border: 0` to things that can be both links and buttons.

File: src/patternfly/components/Divider/divider.scss [hunk 4]
```diff
@@ -1,98 +1,83 @@
...
-  // Vertical
+  @include pf-v5-c-divider--m-horizontal; // default, set to orientation to horizontal
   @include pf-v5-hidden-visible(var(--#{$divider}--Display));
-  @include pf-v5-c-divider--m-horizontal; // set to default to horizontal
 
-  flex-direction: var(--#{$divider}--FlexDirection);
-  flex-shrink: 0;
-  align-items: center;
-  align-self: stretch;
+  display: flex;
```
Comment (mcoker) [inline src/patternfly/components/Divider/divider.scss:34]:
```suggestion
  display: var(--#{$divider}--Display);
```

**Review by mcoker — APPROVED** [review]
LGTM, just a few nits.

**Review by mcoker — APPROVED** [review]
🚀

**Review by srambach — APPROVED** [review]
LGTM - thanks!

**Pull request merged**